### PR TITLE
Add default implementation of IVsSolutionRestoreService4

### LINKED ISSUE #10908: Add a default implementation for IVsSolutionRestoreService4
This would allow https://github.com/dotnet/project-system/pull/7199 to complete, as they'll likely acquire the service via MEF, but currently no one exposes that service.

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Home/issues/10908

Regression? Last working version:

## Description

This would allow dotnet/project-system#7199 to complete, as they'll likely acquire the service via MEF, but currently no one exposes that service.

The real implementation will follow in https://github.com/NuGet/Home/issues/10678. 

## PR Checklist

- [ ] PR has a meaningful title
- [ ] PR has a linked issue.
- [ ] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [ ] N/A

### CODE REVIEW

**Comment by nkolev92** [comment]
cc @ocallesp

This would need to inserted from NuGet side, to get your impl to work.

File: src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs [hunk 1]
```diff
@@ -321,5 +323,11 @@ internal static DependencyGraphSpec CreateMinimalDependencyGraphSpec(string proj
 
             return dgSpec;
         }
+
+        public Task RegisterRestoreInfoSourceAsync(IVsProjectRestoreInfoSource restoreInfoSource, CancellationToken cancellationToken)
+        {
+            return Task.CompletedTask;
```
Comment (erdembayar) [inline src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs:329]: Just one question.
When we use `throw not ImplementedException` vs `Task.CompletedTask` ?

Comment (nkolev92) [reply src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs:329]: This is completely temporary. I don't want the tests on project-system end to throw exceptions. It's up to NuGet to use the project-system in this case.

**Review by erdembayar — APPROVED** [review]

File: src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs [hunk 2]
```diff
@@ -321,5 +323,11 @@ internal static DependencyGraphSpec CreateMinimalDependencyGraphSpec(string proj
...
+            // TODO: https://github.com/NuGet/Home/issues/10678
```
Comment (dominoFire) [inline src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs:330]: Are you going to remove this TODO ?

File: src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs [hunk 3]
```diff
@@ -34,7 +35,8 @@ namespace NuGet.SolutionRestoreManager
     [Export(typeof(IVsSolutionRestoreService))]
     [Export(typeof(IVsSolutionRestoreService2))]
     [Export(typeof(IVsSolutionRestoreService3))]
-    public sealed class VsSolutionRestoreService : IVsSolutionRestoreService, IVsSolutionRestoreService2, IVsSolutionRestoreService3
+    [Export(typeof(IVsSolutionRestoreService4))]
```
Comment (dominoFire) [inline src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs:38]: Do we need a brokered test here?

**Review by dominoFire — APPROVED** [review]
Signing-off to unblock. Please review TODO comment.

File: src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs (hunk 2 above)
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs:330]: The TODO is there to indicate that it's happening later.

File: src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs (hunk 3 above)
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs:38]: This is not a brokered service. Unfortunately we're short on tests here, the manual tests + RPS are what tests this.

**Pull request merged**